x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/search_experience/build_search_ui_config.test.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SchemaTypes } from '../../../../shared/types';
+
+import { buildSearchUIConfig } from './build_search_ui_config';
+
+describe('buildSearchUIConfig', () => {
+  it('builds a configuration object for Search UI', () => {
+    const connector = {};
+    const schema = {
+      foo: 'text' as SchemaTypes,
+      bar: 'number' as SchemaTypes,
+    };
+
+    const config = buildSearchUIConfig(connector, schema);
+    expect(config.apiConnector).toEqual(connector);
+    expect(config.searchQuery.result_fields).toEqual({
+      bar: {
+        raw: {},
+        snippet: {
+          fallback: true,
+          size: 300,
+        },
+      },
+      foo: {
+        raw: {},
+        snippet: {
+          fallback: true,
+          size: 300,
+        },
+      },
+    });
+  });
```

Comment:
Can we add one more super quick test here to cover the `|| {}` branch on this file?

```suggestion
  });

  it('falls back gracefully if schema is undefined', () => {
    const config = buildSearchUIConfig({});
    expect(config.apiConnector).toEqual({});
    expect(config.searchQuery.result_fields).toEqual({});
  });
```

Reply:
Actually looking back at the param/type definitions I'm wondering if this test is accurate:

- If it's possible to pass a falsy/undefined `schema` param, we should indicate that on line 9: `buildSearchUIConfig = (apiConnector: object, schema?: Schema)`
- If it's not possible to pass a falsy/undefined `schema` param, we should get rid of the the `|| {}` fall back?

Thoughts? Am I missing something here?

Reply:
Ohhhh, let me take a closer look.

Reply:
6b3f10b700a6fa9aa745a5cff098a438b88460e1